electrumsv/nodeapi.py

```diff
@@ -557,6 +580,209 @@ async def jsonrpc_getnewaddress_async(request: web.Request, request_id: RequestI
+    parameter_values = transform_parameters(request_id, [ "minconf", "maxconf", "addresses",
+        "include_unsafe" ], parameters)
+
+    minimum_confirmations = 1
+    if len(parameter_values) > 0 and parameter_values[0] is not None:
+        # Incompatibility: It is not necessary to do a `type_check_argument` as the node does.
+        minimum_confirmations = get_integer_parameter(request_id, parameter_values[0])
+
+    maximum_confirmations = 9999999
+    if len(parameter_values) > 1 and parameter_values[1] is not None:
+        # Incompatibility: It is not necessary to do a `type_check_argument` as the node does.
+        maximum_confirmations = get_integer_parameter(request_id, parameter_values[1])
+
+    filter_addresses: set[Address]|None = None
+    if len(parameter_values) > 2 and parameter_values[2] is not None:
+        # Compatibility: Returns a `RPC_TYPE_ERROR` response if the parameter is not a list.
+        type_check_argument(request_id, parameter_values[2], list)
+
+        filter_addresses = set()
+        for entry_value in parameter_values[2]:
+            address_text = get_string_parameter(request_id, entry_value)
+            try:
+                address = Address.from_string(address_text, Net.COIN)
+            except ValueError as value_error:
+                # Compatibility: Raises RPC_INVALID_ADDRESS_OR_KEY if the address is invalid.
+                # rpcwallet.cpp:sendtoaddress (effective check)
+                # Note that we show our error message text not what the node says so that we can
+                # give hints like "wrong verbyte" or "p2sh not accepted" for direct context.
+                raise web.HTTPInternalServerError(headers={ "Content-Type": "application/json" },
+                    text=json.dumps(ResponseDict(id=request_id, result=None,
+                        error=ErrorDict(code=INVALID_ADDRESS_OR_KEY,
+                            message=f"Invalid Bitcoin address: {value_error}"))))
+
+            if address in filter_addresses:
+                # Compatibility: Raises RPC_INVALID_PARAMETER if the address is invalid.
+                # rpcwallet.cpp:sendtoaddress (effective check)
+                raise web.HTTPInternalServerError(headers={ "Content-Type": "application/json" },
+                    text=json.dumps(ResponseDict(id=request_id, result=None,
+                        error=ErrorDict(code=INVALID_PARAMETER,
+                            message=f"Invalid parameter, duplicated address: {entry_value}"))))
+            filter_addresses.add(address)
+
+    only_safe = True
+    if len(parameter_values) > 3 and parameter_values[3] is not None:
+        include_unsafe = get_bool_parameter(request_id, parameter_values[3])
+        only_safe = not include_unsafe
+
+    class NodeUnspentOutputDict(TypedDict):
+        # The UTXO outpoint.
+        txid: str
+        vout: int
+
+        # The address BUT ONLY IF the caller is filtering for it specifically.
```

Comment:
Does this differ from the node implementation? I think the node implementation includes the `address` field whether or not the addresses are being filtered for.

So it seems like it should either be documented in the node-wallet-api.rst incompatibilities or made to be consistent.

I will still go ahead and approve the PR though - not going to block it just for this.